src/Functions/ReplaceStringImpl.h

```diff
@@ -292,7 +307,15 @@ struct ReplaceStringImpl
             const size_t cur_replacement_length = replacement_offsets[i] - prev_replacement_offset - 1;
 
             if (cur_needle_length == 0)
-                throw Exception(ErrorCodes::ARGUMENT_OUT_OF_BOUND, "Length of the pattern argument in function {} must be greater than 0.", name);
+            {
+                copyToOutput(cur_haystack_data, cur_haystack_length, res_data, res_offset);
+                res_offsets[i] = res_offset + 1; // count the zero byte at the end of the string
```

Comment:
The zero byte should be copied too. See the end of this function:
https://github.com/ClickHouse/ClickHouse/blob/04bb11fd81d44f85683ba33559a24b83a1406b27/src/Functions/ReplaceStringImpl.h#L348-L350
If there is no match, then `last_match == nullptr` and we copy `(cur_haystack_end - cur_haystack_data + 1)`, i.e. `cur_haystack_length`.

Reply:
Got it.